packages/luigis-box/src/Component/LuigisBox/LuigisBoxClient.php

```diff
@@ -222,21 +245,42 @@ protected function addAutocompleteSpecificParametersToUrl(
     }
 
     /**
-     * @param string $url
      * @param \Shopsys\LuigisBoxBundle\Model\Batch\LuigisBoxBatchLoadData $luigisBoxBatchLoadData
-     * @param array<string, int> $limitsByType
-     * @return string
+     * @return array
      */
-    protected function addRecommendationsParameters(
-        string $url,
+    protected function getBody(
         LuigisBoxBatchLoadData $luigisBoxBatchLoadData,
-        array $limitsByType,
-    ): string {
+    ): array {
+        $body = [];
+
+        if ($luigisBoxBatchLoadData instanceof LuigisBoxRecommendationBatchLoadData) {
+            $body = $this->addRecommendationSpecificParametersToBody($body, $luigisBoxBatchLoadData);
+        }
+
+        return [$body];
```

Comment:
if the condition above is not met, the method will return an array in an array, that is correct? `[[]]`? In `getData` method, there is a condition `if ($body !== [])` so I suppose you do not want to return `[[]]` here. What about this?

```
$body = [];

if ($luigisBoxBatchLoadData instanceof LuigisBoxRecommendationBatchLoadData) {
    $body = [$this->addRecommendationSpecificParametersToBody($body, $luigisBoxBatchLoadData)];
}

return $body;
```